relay-metrics/src/aggregatorservice.rs

```diff
@@ -128,144 +94,250 @@ pub struct FlushBuckets {
+    ///
+    /// If set, buckets are partitioned by (bucket key % flush_partitions), and routed
+    /// by setting the header `X-Sentry-Relay-Shard`.
+    pub flush_partitions: Option<u64>,
+}
+
+impl Default for AggregatorServiceConfig {
+    fn default() -> Self {
+        Self {
+            max_flush_bytes: 5_000_000, // 5 MB
+            flush_partitions: None,
+            max_total_bucket_bytes: None,
+            aggregator: AggregatorConfig::default(),
+        }
+    }
+}
+
+/// Contains an [`AggregatorServiceConfig`] for a specific scope.
+///
+/// For now, the only way to scope an aggregator is by [`MetricNamespace`].
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct ScopedAggregatorConfig {
+    /// Name of the aggregator, used to tag statsd metrics.
+    pub name: String,
+    /// Condition that needs to be met for a metric or bucket to be routed to a
+    /// secondary aggregator.
+    pub condition: Condition,
+    /// The configuration of the secondary aggregator.
+    pub config: AggregatorServiceConfig,
+}
+
+/// Condition that needs to be met for a metric or bucket to be routed to a
+/// secondary aggregator.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(tag = "op", rename_all = "lowercase")]
+pub enum Condition {
+    /// Checks for equality on a specific field.
+    Eq(Field),
+}
+
+/// Defines a field and a field value to compare to when a [`Condition`] is evaluated.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(tag = "field", content = "value", rename_all = "lowercase")]
+pub enum Field {
+    /// Field that allows comparison to a metric or bucket's namespace.
+    Namespace(MetricNamespace),
+}
+
 enum AggregatorState {
     Running,
     ShuttingDown,
 }
 
-/// A service for aggregationg metric buckets.
-///
-/// Buckets are flushed to a receiver after their time window and a grace period have passed.
-/// Metrics with a recent timestamp are given a longer grace period than backdated metrics, which
-/// are flushed after a shorter debounce delay. See [`AggregatorServiceConfig`] for configuration options.
-///
-/// Internally, the aggregator maintains a continuous flush cycle every 100ms. It guarantees that
-/// all elapsed buckets belonging to the same [`ProjectKey`] are flushed together.
+/// Service that routes metrics & metric buckets to the appropriate aggregator.
 ///
-/// Receivers must implement a handler for the [`FlushBuckets`] message.
+/// Each aggregator gets its own configuration.
+/// Metrics are routed to the first aggregator which matches the configuration's [`Condition`].
+/// If no condition matches, the metric/bucket is routed to the `default_aggregator`.
 pub struct AggregatorService {
-    aggregator: aggregator::Aggregator,
-    state: AggregatorState,
-    receiver: Option<Recipient<FlushBuckets, NoResponse>>,
+    default_aggregator: aggregator::Aggregator,
+    secondary_aggregators: BTreeMap<MetricNamespace, aggregator::Aggregator>,
```

Comment:
Part of the motivation for this refactor was to have a "metrics router light" in the pre-aggregator that wraps multiple aggregators, instead of having the suboptimal "lossless" pre-aggregator. This PR should already introduce that wrapper type IMO.

Reply:
maybe we can reconsider this? I agree with @iker-barriocanal  that this PR is already pretty big, and I'd say the current refactor stands well as a PR on its own, where splitting the 'router' and 'routerservice' would be a good PR by itself

Reply:
should have updated the comment, the PR includes the router already!

Reply:
Right, sorry! I was looking at the PR comment without checking the latest changes.